core/src/main/java/org/apache/iceberg/SystemProperties.java

```diff
@@ -41,7 +41,10 @@ private SystemProperties() {}
 
   public static final int IO_MANIFEST_CACHE_MAX_FILEIO_DEFAULT = 8;
 
-  static boolean getBoolean(String systemProperty, boolean defaultValue) {
+  public static final String DELETE_POS_FILES_THREADS_ENABLED =
```

Comment:
I think this should be an engine option rather than a System property. @aokolnychyi what do you think?

Reply:
Using sys properties similar to worker threads in ThreadPools codebase. Making engine option wasn't straight forward via Deletes codebase. It may be easier to start with sys properties (similar to worker threads) and move to other mode if needed later?.